Azure/iotedge
# fix identity cleanup

It is possible to have identities get orphaned from modules when modules are unable to be brought up by EA during deployment. In order to fix this, a check for orphaned identities is needed.

This functionality will be feature-flagged behind "EnableOrphanedIdentityCleanup". Default is false/disabled, with setting to true enabling the feature. Below is example usage:

```json
 "env": {
    "EnableOrphanedIdentityCleanup": {
      "value": "true"
    }
  },
```

Manual test result:
1. Deploy only EA + EH
2. Deploy EA + EH with 40 bad modules - See NA in portal
3. Deploy EA + EH (new bits) with 40 bad modules - puts EA in a state where identities can be cleaned up
4. Deploy EA + EH (new bits) with 40 new modules - See only good modules, NA are gone

PSA:
This change does have a side-effect of possibly removing Host-Level modules; this is due to improper handling of managedBy elsewhere that in conjunction with this change causes modules not managed by edgeAgent to also get affected.
